RecoTracker/TkDetLayers/src/TIBRing.h

```diff
@@ -49,10 +51,8 @@ class TIBRing final : public GeometricSearchDet {
 
   // methods for groupedCompatibleDets implementation
   struct SubRingCrossings {
-    SubRingCrossings() : isValid_(false){};
-    SubRingCrossings(int ci, int ni, float nd) : isValid_(true), closestIndex(ci), nextIndex(ni), nextDistance(nd) {}
+    SubRingCrossings(int ci, int ni, float nd) : closestIndex(ci), nextIndex(ni), nextDistance(nd) {}
 
-    bool isValid_;
     int closestIndex;
     int nextIndex;
     float nextDistance;
```

Comment:
I was looking at https://www.cppstories.com/2018/05/using-optional/ and didn't find particular concerns, apart for not being very sure about the memory use.

If I understand correctly, the old variant has a size of `4*4` (instead of the nominal `1+3*4`) because bool is listed first and the compiler will likely pad it?
And the optional would similarly replace the equivalent initialization flag with the 4-aligned value.

Just to be more "sure", for the following 
```C++
struct E3o {bool f; int a; int b; float c;} e3o;
struct E3 {int a; int b; float c;};
std::optional<E3> e3opt;
std::cout<<sizeof(e3o)<<" "<<sizeof(e3opt)<<std::endl;
```
I get `16 16`.

Reply:
> If I understand correctly, the old variant has a size of `4*4` (instead of the nominal `1+3*4`) because bool is listed first and the compiler will likely pad it?

Correct. (it would be padded even if the `bool` would be last)

> And the optional would similarly replace the equivalent initialization flag with the 4-aligned value.

Correct, as your test shows.

Would profiling tests be worth it?

Reply:
I played a bit in godbolt https://godbolt.org/z/j3oP95f9Y (not sure if my test is still too simplistic). It seems to me that GCC would emit a bit less efficient code for `std::optional` (includes copying stuff between stack and registers), but clang would be able to emit nearly identical code for `std::optional`.

Reply:
still, once inlined in ```bar``` the code is optimal.